File: web/src/application/use-cases/list-documents.ts
/**
 * ListDocumentsUseCase — fetch the user's KB documents (FR-31).
 *
 * Thin wrapper over `DocumentsApiPort.list` so the application layer
 * exposes a uniform "use case" surface for hooks/components to consume
 * via the composition root.
 */

import type {
  DocumentListFilters,
  DocumentsApiPort,
} from "@/application/ports/documents-api.port";
import type { DocumentListItem } from "@/domain/entities/document";

export class ListDocumentsUseCase {
  constructor(private readonly api: DocumentsApiPort) {}

  execute(filters?: DocumentListFilters): Promise<DocumentListItem[]> {
    return this.api.list(filters);
  }
}

File: python_backend/app/application/use_cases/list_documents.py
"""List documents for the current user."""
from typing import Optional

from app.application.ports.documents_repository import DocumentsRepository
from app.domain.entities.document import Document


class ListDocumentsUseCase:
    def __init__(self, repo: DocumentsRepository) -> None:
        self.repo = repo

    def execute(
        self,
        *,
        user_id: str,
        scenario: Optional[str] = None,
        doc_type: Optional[str] = None,
    ) -> list[Document]:
        return self.repo.list(user_id, scenario=scenario, doc_type=doc_type)
